src/stan/model/model_header.hpp

```diff
@@ -8,13 +8,13 @@
 #include <stan/math/prim/mat/fun/value_of_rec.hpp>
 #include <stan/math/prim/mat/meta/get.hpp>
 
+#include <stan/math/prim/scal.hpp>
```

Comment:
the order here was intentional. Please change it back to: arr then mat then scal.